setup.py

```diff
@@ -92,7 +92,10 @@ def run(self):
     def build_extension(self, ext):
         extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
 
-        if in_git():
+        # Init & update all submodules if not already (the user might be pinned
+        # on some particular commit or have working tree changes, don't destroy
+        # those)
+        if in_git() and not os.path.exists(".git/modules"):
```

Comment:
I like the idea of this -- I have resorted to some pretty hacky hacks to get git to not update submodules (adding a random space or newline to a file and then not committing it being my favorite).

However, I also like that currently when people pull a new build, they don't have to worry about updating submodules.  Would adding a `--no-update-modules` flag be sufficient (I am about to submit a PR adding a `--cmake-args` flag, so I can add it to that)?

CC: @msavva

Reply:
I see your point. Yes that would work too, I would just need to remember to always add that flag (which is okay) :)

A more general question from a `setup.py` noob -- do setuptools have something similar to cmake's cache? Like, you enable / set up some flags and properties and they get remembered for the next invocation; so you don't have to specify the whole command-line but can do just `./setup.py` and it will pick up the rest from *somewhere*, which is (like cmake's build dir) not commited to the git repository but private to your working copy? Or am I asking for the impossible?

Reply:
Yeah... unfortunately setup.py doesn't do that.  We could roll our own as pickling out `args` and reloading if it exists would work.....

Reply:
I also like the idea.  I agree with @erikwijmans 's suggestion to make this into a `--no-update-modules` flag. It's good to accommodate for the expectation that everything should work through a naive pull-and-build.  I'm not aware of a way to make `setup.py` cache flags etc. :(

Reply:
Made a PR into this PR with a go at caching (it contains more than just that tho).